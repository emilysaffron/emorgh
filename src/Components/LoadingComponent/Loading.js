import getGridGroups from "../../Helpers/GetGridGroups";
import Grid from "@bbc/psammead-grid";
import styled from "@emotion/styled";
import Loader from "react-loader-spinner";
const StyledLoader = styled(Loader)`
  margin-top: 300px;
`;

const Loading = () => {
  return (
    <Grid
      enableGelGutters
      columns={getGridGroups([7, 7, 7, 7, 7, 7])}
      margins={getGridGroups([true, true, true, true, true, true])}
    >
      <Grid
        item
        startOffset={getGridGroups([4, 4, 4, 4, 4, 4])}
        columns={getGridGroups([8, 8, 8, 8, 8, 8])}
      >
        <StyledLoader
          type="ThreeDots"
          color="lightgray"
          height={80}
          width={80}
        />
      </Grid>
    </Grid>
  );
};

export default Loading;
