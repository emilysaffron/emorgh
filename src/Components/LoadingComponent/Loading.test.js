import Loading from "./Loading";
import renderer from "react-test-renderer";

it("Loading Snapshot", () => {
  const tree = renderer.create(<Loading />).toJSON();
  expect(tree).toMatchSnapshot();
});
