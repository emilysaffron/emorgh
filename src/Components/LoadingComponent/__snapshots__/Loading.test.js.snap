// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Snapshot 1`] = `
<div
  className="css-9oqp92-GridComponent e57qer20"
  dir="ltr"
>
  <div
    className="css-hfalua-GridComponent e57qer20"
    dir="ltr"
  >
    <div
      aria-busy="true"
      className=" css-r939z9"
      style={Object {}}
    >
      <svg
        aria-label="audio-loading"
        fill="lightgray"
        height={80}
        viewBox="0 0 120 30"
        width={80}
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="15"
          cy="15"
          r={15}
        >
          <animate
            attributeName="r"
            begin="0s"
            calcMode="linear"
            dur="0.8s"
            from="15"
            repeatCount="indefinite"
            to="15"
            values="15;9;15"
          />
          <animate
            attributeName="fillOpacity"
            begin="0s"
            calcMode="linear"
            dur="0.8s"
            from="1"
            repeatCount="indefinite"
            to="1"
            values="1;.5;1"
          />
        </circle>
        <circle
          attributeName="fillOpacity"
          cx="60"
          cy="15"
          from="1"
          r={9}
          to="0.3"
        >
          <animate
            attributeName="r"
            begin="0s"
            calcMode="linear"
            dur="0.8s"
            from="9"
            repeatCount="indefinite"
            to="9"
            values="9;15;9"
          />
          <animate
            attributeName="fillOpacity"
            begin="0s"
            calcMode="linear"
            dur="0.8s"
            from="0.5"
            repeatCount="indefinite"
            to="0.5"
            values=".5;1;.5"
          />
        </circle>
        <circle
          cx="105"
          cy="15"
          r={15}
        >
          <animate
            attributeName="r"
            begin="0s"
            calcMode="linear"
            dur="0.8s"
            from="15"
            repeatCount="indefinite"
            to="15"
            values="15;9;15"
          />
          <animate
            attributeName="fillOpacity"
            begin="0s"
            calcMode="linear"
            dur="0.8s"
            from="1"
            repeatCount="indefinite"
            to="1"
            values="1;.5;1"
          />
        </circle>
      </svg>
    </div>
  </div>
</div>
`;
